Makefile

```diff
@@ -21,36 +21,36 @@ include Makefile.common
 .PHONY: help
 help: ## Display this help and any documented user-facing targets. Other undocumented targets may be present in the Makefile.
 help:
-	@awk 'BEGIN {FS = ": ##"; printf "Usage:\n  make <target>\n\nTargets:\n"} /^[a-zA-Z0-9_\.\-\/%]+: ##/ { printf "  %-45s %s\n", $$1, $$2 }' $(MAKEFILE_LIST)
+	@awk 'BEGIN {FS = ": ##"; printf "Usage:\n  make <target>\n\nCommon Targets:\n"} /^[a-zA-Z0-9_\.\-\/%]+: ##/ { printf "  %-45s %s\n", $$1, $$2 } END { printf "\nFor development workflow:\n  make dev              # Complete development workflow\n  make check            # Run all code quality checks\n  make test             # Run all tests\n  make precheck         # Quick pre-commit checks\n\nFor building:\n  make                  # Build TiDB server (default)\n  make server           # Build TiDB server\n  make clean            # Clean build artifacts\n" }' $(MAKEFILE_LIST)
```

Comment:
This does improve the usability, but the hard-coded target (`make dev`, `make check`, ...) may get forgotton to change when we rename them. I prefer we only keep `Common Targets:` parts

Reply:
addressed